sdk/ai/azure-ai-resources/azure/ai/resources/entities/connection_subtypes.py

```diff
@@ -99,16 +100,31 @@ def set_current_environment(self, credential: Optional[TokenCredential] = None):
         :type credential: :class:`~azure.core.credentials.TokenCredential`
         """
 
-        import os
-        def get_api_version_case_insensitive(connection):
-            if connection.api_version == None:
-                raise ValueError(f"Connection {connection.name} is being used to set environment variables, but lacks required api_version")
-            return connection.api_version.lower()
+        import pkg_resources
```

Comment:
FYI: it seems like `pkg_resources` is deprecated: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html

Reply:
Thx for the reminder. Fixed.